cloud/pkg/devicecontroller/controller/downstream.go

```diff
@@ -616,13 +616,13 @@ func (dc *DownstreamController) deviceUpdated(device *v1alpha2.Device) {
 		if isDeviceUpdated(cachedDevice, device) {
 			// if node selector updated delete from old node and create in new node
 			if isNodeSelectorUpdated(cachedDevice.Spec.NodeSelector, device.Spec.NodeSelector) {
-				dc.deviceAdded(device)
 				deletedDevice := &v1alpha2.Device{ObjectMeta: cachedDevice.ObjectMeta,
 					Spec:     cachedDevice.Spec,
 					Status:   cachedDevice.Status,
 					TypeMeta: device.TypeMeta,
 				}
 				dc.deviceDeleted(deletedDevice)
+				dc.deviceAdded(device)
```

Comment:
Looks like the code here will only be called when the device is updated, and from the pr description, it's related to the device creation?

Reply:
In our usage scenario, we will create the device first, and then bind the node. Therefore, the device will be added first, but the device is updated due to the binding node, so the code for updating the device will be entered. 

The `deviceAdded` and `deviceDeleted` here both use the **deviceName** to operate `sync.Map`, which leads to the problem that the device cannot be queried after the node is bound.

Reply:
Sounds good.